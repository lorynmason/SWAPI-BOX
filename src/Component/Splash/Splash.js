import React from 'react';
import '../styles/main.scss';
import PropTypes from 'prop-types';

const Splash = ({exitSplash, films}) => {
  const randomNum = Math.floor(Math.random() * Math.floor(6))
  const film = films[randomNum]
  console.log(film)
  if(film) {
    return (
      <section className='background'>
            <div className="fade">
              <div className="logo">
                <img alt='StarWars logo'
                    onClick={() => exitSplash()} 
                    src="http://pngimg.com/uploads/star_wars_logo/star_wars_logo_PNG29.png" title="click to enter site"/> 
              </div>
              </div>
              <div className="crawl">
                {film.text}
              <p>{film.title}</p>
              <p>{film.date}</p>
              </div>
              <div className="button-div">
              <button onClick={() => exitSplash()}className='enter-btn'>
              ENTER </button>
              </div>
            </section>
    )
  } else {
    
    return (
      <section className='background'>
        <div className="fade">
        </div>
      </section>
    )
  }
}

Splash.propTypes = {
exitSplash: PropTypes.func.isRequired,
films: PropTypes.array.isRequired
}

export default Splash;
