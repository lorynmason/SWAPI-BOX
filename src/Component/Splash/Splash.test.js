import React from 'react'
import { shallow, mount } from 'enzyme'
import Splash from './Splash'

describe('Splash', () => {
  let wrapper
  let mockExitSplash
  let mockfilms

  beforeEach(() => {
    mockExitSplash = jest.fn()
    mockfilms = [{
      title: 'A New Hope',
      text: 'A Long Time Ago...',
      date: '1976'
    }, {
      title: 'A New Hope',
      text: 'A Long Time Ago...',
      date: '1976'
    }, {
      title: 'A New Hope',
      text: 'A Long Time Ago...',
      date: '1976'
    }, {
      title: 'A New Hope',
      text: 'A Long Time Ago...',
      date: '1976'
    }, {
      title: 'A New Hope',
      text: 'A Long Time Ago...',
      date: '1976'
    }, {
      title: 'A New Hope',
      text: 'A Long Time Ago...',
      date: '1976'
    }, {
      title: 'A New Hope',
      text: 'A Long Time Ago...',
      date: '1976'
    }]
    wrapper = mount(<Splash changePage={mockExitSplash} splash={mockfilms} />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('logo should call mockExitSplash on click', () => {
    const logoImg = wrapper.find('img')
    logoImg.simulate('click')

    expect(mockExitSplash).toHaveBeenCalled()
  })

  it('Enter button should call mockExitSplash on click', () => {
    const button = wrapper.find('[type="button"]')
    button.simulate('click')

    expect(mockExitSplash).toHaveBeenCalled()
  })

  it('Should render backgound if no film', () => {
    mockfilms = []
    wrapper = shallow(<Splash changePage={mockExitSplash} splash={mockfilms} />)
    expect(wrapper).toMatchSnapshot()
  })
})
