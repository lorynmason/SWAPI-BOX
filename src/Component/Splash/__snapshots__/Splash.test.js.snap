// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Splash should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Splash
    exitSplash={[MockFunction]}
    film={
      Object {
        "date": "1976",
        "text": "A Long Time Ago...",
        "title": "A New Hope",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="fade"
        >
          <div
            className="logo"
          >
            <img
              alt="StarWars logo"
              onClick={[Function]}
              src="http://pngimg.com/uploads/star_wars_logo/star_wars_logo_PNG29.png"
            />
          </div>
        </div>,
        <div
          className="crawl"
        >
          A Long Time Ago...
          <p>
            A New Hope
          </p>
          <p>
            1976
          </p>
        </div>,
        <button
          className="enter-btn"
          onClick={[Function]}
        >
          ENTER 
        </button>,
      ],
      "className": "background",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="logo"
          >
            <img
              alt="StarWars logo"
              onClick={[Function]}
              src="http://pngimg.com/uploads/star_wars_logo/star_wars_logo_PNG29.png"
            />
          </div>,
          "className": "fade",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="StarWars logo"
              onClick={[Function]}
              src="http://pngimg.com/uploads/star_wars_logo/star_wars_logo_PNG29.png"
            />,
            "className": "logo",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "StarWars logo",
              "onClick": [Function],
              "src": "http://pngimg.com/uploads/star_wars_logo/star_wars_logo_PNG29.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "A Long Time Ago...",
            <p>
              A New Hope
            </p>,
            <p>
              1976
            </p>,
          ],
          "className": "crawl",
        },
        "ref": null,
        "rendered": Array [
          "A Long Time Ago...",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "A New Hope",
            },
            "ref": null,
            "rendered": "A New Hope",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "1976",
            },
            "ref": null,
            "rendered": "1976",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "ENTER ",
          "className": "enter-btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "ENTER ",
        "type": "button",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="fade"
          >
            <div
              className="logo"
            >
              <img
                alt="StarWars logo"
                onClick={[Function]}
                src="http://pngimg.com/uploads/star_wars_logo/star_wars_logo_PNG29.png"
              />
            </div>
          </div>,
          <div
            className="crawl"
          >
            A Long Time Ago...
            <p>
              A New Hope
            </p>
            <p>
              1976
            </p>
          </div>,
          <button
            className="enter-btn"
            onClick={[Function]}
          >
            ENTER 
          </button>,
        ],
        "className": "background",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="logo"
            >
              <img
                alt="StarWars logo"
                onClick={[Function]}
                src="http://pngimg.com/uploads/star_wars_logo/star_wars_logo_PNG29.png"
              />
            </div>,
            "className": "fade",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="StarWars logo"
                onClick={[Function]}
                src="http://pngimg.com/uploads/star_wars_logo/star_wars_logo_PNG29.png"
              />,
              "className": "logo",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "StarWars logo",
                "onClick": [Function],
                "src": "http://pngimg.com/uploads/star_wars_logo/star_wars_logo_PNG29.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "A Long Time Ago...",
              <p>
                A New Hope
              </p>,
              <p>
                1976
              </p>,
            ],
            "className": "crawl",
          },
          "ref": null,
          "rendered": Array [
            "A Long Time Ago...",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "A New Hope",
              },
              "ref": null,
              "rendered": "A New Hope",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "1976",
              },
              "ref": null,
              "rendered": "1976",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "ENTER ",
            "className": "enter-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "ENTER ",
          "type": "button",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
